tachyon/zk/plonk/expressions/plonk_expression.h

```diff
@@ -0,0 +1,73 @@
+// Copyright 2020-2022 The Electric Coin Company
+// Copyright 2022 The Halo2 developers
+// Use of this source code is governed by a MIT/Apache-2.0 style license that
+// can be found in the LICENSE-MIT.halo2 and the LICENCE-APACHE.halo2
+// file.
+
+#ifndef TACHYON_ZK_PLONK_EXPRESSIONS_PLONK_EXPRESSION_H_
+#define TACHYON_ZK_PLONK_EXPRESSIONS_PLONK_EXPRESSION_H_
+
+#include <memory>
+#include <optional>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include "tachyon/base/containers/container_util.h"
+#include "tachyon/base/logging.h"
+#include "tachyon/base/ref.h"
+#include "tachyon/zk/expressions/expression.h"
+
+namespace tachyon::zk::plonk {
+
+template <typename F>
+class ExpressionFactory;
+
+template <typename F, typename T>
+class Evaluator;
+
+template <typename F>
+class SelectorExpression;
+
+template <typename F>
+class FixedExpression;
+
+template <typename F>
+class AdviceExpression;
+
+template <typename F>
+class InstanceExpression;
+
+template <typename F>
+class ChallengeExpression;
+
+// A Expression represents a polynomial.
+template <typename F>
+class PlonkExpression : public Expression<F> {
```

Comment:
Isn't it okay for `Expression<F>` to have all the methods here, it's safe based on the type check, the type check doesn't need plonk specific things.

Reply:
I think that would work since we put all the plonk types back in the `ExpressionType`. Will do!